### PR TITLE
Fixes loading the neighbours of a relation that plays a role in another relation

### PR DESCRIPTION
**Pull request by sorsaffari**
## What is the goal of this PR?
Relations that play a role in other relations can now be double-clicked to display their roleplayers. The neighbour offset (initially set to neighbours limit) is now only set on relations that are a target of the _Load Roleplayers_ option, as opposed to all relations.

## What are the changes implemented in this PR?
resolves https://github.com/graknlabs/workbase/issues/302

### CODE REVIEW

File: src/renderer/components/shared/CanvasDataBuilder.js [hunk 1]
```diff
@@ -586,11 +583,19 @@ const buildRPInstances = async (answers, currentData, shouldLimit, graknTx) => {
   data.nodes = data.nodes.filter((node, index, self) => index === self.findIndex(t => t.id === node.id));
   data.edges = data.edges.filter((edge, index, self) => index === self.findIndex(t => t.id === edge.id));
 
-  // exclude any nodes and edges that have already been constructed and visualised (i.e. currentData)
   if (currentData) {
+    // exclude any nodes and edges that have already been constructed and visualised (i.e. currentData)
```
Comment (flyingsilverfin) [inline src/renderer/components/shared/CanvasDataBuilder.js:587]: thanks for keeping comments, very useful

File: src/renderer/components/shared/CanvasDataBuilder.js [hunk 2]
```diff
@@ -132,6 +132,7 @@ const buildCommonInstanceNode = (instance, graqlVar, explanation, queryPattern)
   node.baseType = instance.baseType;
   node.var = graqlVar;
   node.attrOffset = 0;
+  node.offset = 0;
```
Comment (flyingsilverfin) [inline src/renderer/components/shared/CanvasDataBuilder.js:135]: hang on, what is this offset and why is it not set anymore?

Comment (sorsaffari) [reply src/renderer/components/shared/CanvasDataBuilder.js]: we no longer have the `offset` property on the node object. addressed in https://github.com/graknlabs/workbase/pull/304/commits/27f361b79d6deadfb1d482f13d17f92a9c5359b0

**Comment by sorsaffari** [comment]
> performance is comparable to before right?

feels a little slower, but not quite noticeable

File: src/renderer/components/Visualiser/store/actions.js [hunk 3]
```diff
@@ -88,11 +88,26 @@ export default {
     try {
       commit('loadingQuery', true);
       const graknTx = global.graknTx[rootState.activeTab];
-      const filteredResult = await getFilteredNeighbourAnswers(visNode, graknTx, neighboursLimit);
+
+      const currentData = {
+        nodes: state.visFacade.getAllNodes(),
+        edges: state.visFacade.getEdge(),
```
[the remark this answers is not in the document]
Comment (sorsaffari) [reply src/renderer/components/Visualiser/store/actions.js]: ahhh, it's already there. I just needed to use it.

addressed in https://github.com/graknlabs/workbase/pull/304/commits/70ef4160dd82bb695e49d4fc11eff7b815e8d6fb

**Pull request merged**